Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -202,22 +191,37 @@ - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCha
     }
 }
 
-- (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCursor:(unichar)precedingChar {
-    // Determine the type of the character
-    enum CharacterType {
-        CharacterTypeSeparator = 0,
-        CharacterTypeNormal
-    };
-    enum CharacterType deletedCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:deletedChar] || deletedChar == 0)
-    ? CharacterTypeSeparator
-    : CharacterTypeNormal;
-    enum CharacterType currentCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:precedingChar] || precedingChar == 0)
-    ? CharacterTypeSeparator
-    : CharacterTypeNormal;
+- (void)deleteTypedCharacter:(unichar)deletedChar
+withCharacterNowPrecedingCursor:(unichar)precedingChar
+                    location:(NSUInteger)location
+                textViewText:(nonnull NSString *)textViewText {
+    // Determine the character types
+    enum CharacterType deletedCharacterType = [self characterTypeOfCharacter:deletedChar];
+    enum CharacterType precedingCharacterType = [self characterTypeOfCharacter:precedingChar];
+
+    __strong __auto_type delegate = self.delegate;
 
     switch (self.state) {
         case HKWMentionsStartDetectionStateQuiescentReady: {
-            if (currentCharacterType == CharacterTypeNormal) {
+            BOOL canCreateMention = NO;
```

Comment:
do we wanna do this at the call site for delete, with a function in that class, so we dont have to pass all this extra info in, and just pass in "canCreateMention"

Reply:
If we would want to move the `canCreateMention` logic to caller, we would still have to pass
- location (to begin mentions at that particular location) 
- keyword (to search mentions for that particular word) 

which would not be helping us in reducing the number of parameters to be passed and also as we want to check the character type which can be done better here as we have an enum and helper method in place.